lib/src/pub.dart

```diff
@@ -87,6 +108,41 @@ class Pubspec {
       this._contents,
       this._yamlMap);
 
+  static Future<Pubspec> init() {
+    var completer = new Completer();
+    var packageRoot = p.current;
+    var pubspecPath = p.join(packageRoot, _PUBSPEC);
+    var exp = new RegExp(r"(-|\.).*$");
```

Comment:
I only want to remove `.dart`, `-dart`, `_dart` suffixes,.  The reason is because a lot of dart github repos are named e.g. `quiver-dart`, `which.dart`, `foo_dart`, and when people clone those repos locally the directory has the same name, but the package names do not and should not include the suffix e,g, `quiver` and `which`.

After doing the suffix removal, need to validate that it is a valid identifier i.e. `new RegExp(r'[_a-zA-Z][_a-zA-Z0-9]*').hasMatch(name)`